src/pages/iou/MoneyRequestSelectorPage.js

```diff
@@ -0,0 +1,173 @@
+    const iouType = useRef(lodashGet(props.route, 'params.iouType', ''));
+    const {translate} = useLocalize();
+
+    const isEditing = useRef(lodashGet(props.route, 'path', '').includes('amount'));
+    const reportID = useRef(lodashGet(props.route, 'params.reportID', ''));
+
+    const title = {
+        [CONST.IOU.MONEY_REQUEST_TYPE.REQUEST]: translate('iou.requestMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SEND]: translate('iou.sendMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SPLIT]: translate('iou.splitBill'),
+    };
+    const titleForStep = isEditing.current ? translate('iou.amount') : title[iouType.current];
+
+    const navigateBack = () => {
+        Navigation.goBack(isEditing.current ? ROUTES.getMoneyRequestConfirmationRoute(iouType.current, reportID.current) : null);
+    };
+
+    const [isDraggingOver, setIsDraggingOver] = useState(false);
+
+    return (
+        <FullPageNotFoundView shouldShow={!IOUUtils.isValidMoneyRequestType(iouType.current)}>
+            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+                {({safeAreaPaddingBottomStyle}) => (
+                    <DragAndDrop
+                        dropZoneId={CONST.RECEIPT.DROP_NATIVE_ID}
+                        activeDropZoneId={CONST.RECEIPT.ACTIVE_DROP_NATIVE_ID}
+                        onDragEnter={() => {
+                            setIsDraggingOver(true);
+                        }}
+                        onDragLeave={() => {
+                            setIsDraggingOver(false);
+                        }}
+                        onDrop={(e) => {
+                            e.preventDefault();
+                            setIsDraggingOver(false);
+                            const file = lodashGet(e, ['dataTransfer', 'files', 0]);
+
+                            if (!ReceiptUtils.isValidReceipt(file, props)) {
+                                return;
+                            }
+
+                            const filePath = URL.createObjectURL(file);
+                            IOU.setMoneyRequestReceipt(filePath, file.name);
+                            NavigateToNextIOUPage(props.iou, iouType, reportID, props.report, props.currentUserPersonalDetails);
+                        }}
+                    >
+                        <View
+                            nativeID={CONST.RECEIPT.DROP_NATIVE_ID}
+                            style={[styles.flex1, safeAreaPaddingBottomStyle]}
+                        >
+                            <HeaderWithBackButton
+                                title={titleForStep}
+                                onBackButtonPress={navigateBack}
+                            />
+                            <TabSelector moneyRequestID={`${iouType.current}${reportID.current}`} />
+                            {props.tabSelected === CONST.TAB.TAB_MANUAL ? (
+                                <MoneyRequestAmountPage
+                                    route={props.route}
```

Comment:
Just noticed that `MoneyRequestAmountPage` does not need to take the `route` as a prop anymore. It can just use the `useRoute` hook instead.

Reply:
Can you elaborate? I think we are still using `props.route`?

Reply:
Instead of using the `route` from `props` you can use the `useRoute` hook `const route = useRoute();`. A similar comment is here https://github.com/Expensify/App/pull/23046#discussion_r1272110747 to remove the rest of the props.

Reply:
This prop (and all props) have been removed.